src/NuGet.Clients/VisualStudio/IVsPackageManagerProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// Interface allowing integration of alternate package manager suggestion for a NuGet package. 
+    /// For example jQuery may appear on Bower and it might be more appropriate to install a package from Bower for certain projects 
+    /// </summary>
+    [ComImport]
+    [Guid("BCED5BF2-40FC-4D9F-BF0A-43CD4E9FF65F")]
+    public interface IVsPackageManagerProvider
+    {
+        /// <summary>
+        /// Localized display package manager name.
+        /// </summary>
+        string PackageManagerName { get; }
+
+        /// <summary>
+        /// Package manager unique id.
+        /// </summary>
+        string PackageManagerId { get; }
+
+        /// <summary>
+        /// The tool tip description for the package
+        /// </summary>
+        string Description { get; }
+
+        /// <summary>
+        /// Check if a recommendation should be surfaced for an alternate package manager. 
+        /// This code should not rely on slow network calls, and should return rapidly.
+        /// </summary>
+        /// <param name="packageId">Current package id</param>
+        /// <param name="projectName">Unique project name for finding the project through VS dte</param>
+        /// <param name="token">Cancellation Token</param>
+        /// <returns>return true if need to direct to integrated package manager for this package</returns>
+        Task<bool> CheckForPackage(string packageId, string projectName, CancellationToken token);
```

Comment:
It is a convention to have a suffix 'async' when the return type is Task or Task<T>. Consider changing this to 'CheckForPackageAsync' if possible.